### ISSUE TITLE
[R] Ensure that bit64's "new semantics" work with arrow

### ISSUE THREAD

**Issue by jonkeane** [apache/arrow#44208]
### Describe the enhancement requested

https://github.com/truecluster/bit64/blob/1d34b37c14899dba9a5e915ed5e1fb671926f072/NEWS#L15-L18

Might be as simple as duplicating tests like https://github.com/apache/arrow/blob/0ddee234a0722c75e6b2c3e9aa9bbad1d27267af/r/tests/testthat/test-Array.R#L342-L353 with the option enabled.

### Component(s)

R

**Comment by AkashNegi1** [comment]
could you elaborate on what to do to solve this issue? Do I have to use the new semantics for the above give test code only and test that out???

**Comment by jonkeane** [comment]
> could you elaborate on what to do to solve this issue? Do I have to use the new semantics for the above give test code only and test that out???

Yeah, I would duplicate the test code, turn on the bit64 option in that block and make sure there are no differences as a start.

**Comment by AkashNegi1** [comment]
May I try it?

**Comment by jonkeane** [comment]
Yup, of course. Feel free to tag me on the PR when you've got it ready.

**Comment by jonkeane** [comment]
Issue resolved by pull request 46651
https://github.com/apache/arrow/pull/46651
